packages/vehicle-lifecycle-network/test/vda.js

```diff
@@ -124,4 +124,29 @@ describe('Vehicle Lifecycle Network', function() {
                 });
         });
     });
-});
\ No newline at end of file
+
+    describe('ScrapAllVehiclesByColour', function() {
+        it('should select vehicles by colour and change vehicles status to SCRAPPED', function() {
+            /* vehicle with beige colour
+               and id 123456789 resides
+               in reposritory
+            */
+            var vehicleId = '123456789';
+            var scrapVehicleTransaction = factory.newTransaction(NS_D, 'ScrapAllVehiclesByColour');
+            scrapVehicleTransaction.colour = 'Beige';
+            return businessNetworkConnection.submitTransaction(scrapVehicleTransaction)
```

Comment:
It might be worth adding a precondition check to see that the car you expect to be scrapped is not already scrapped before the transaction runs. If you do, you will find that the precondition check fails as the car has already been scrapped by the previous test. The before() function that does the test set up needs to be changed to beforeEach() to ensure a clean environment before each test.

Reply:
The review comment has been incorporated